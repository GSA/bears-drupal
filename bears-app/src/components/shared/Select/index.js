/**
 * a functional component to create a select from a list of options
 * @component
 * @param {string} label - The inherited options object of value and label
 * @param {string} htmlFor - The inherited component id that the label is for
 * @param {array} options - The inherited options array of objects with value and label keys
 * @return {html} returns a semantic html select element with options
 */
function Select({ label, htmlFor, options }) {
  /**
   * a functional component to create a label for options.
   * @function
   * @param {string} label - The inherited lable value
   * @param {string} htmlFor - The inherited component id that the label is related to
   * @return {html} returns a semantic html label
   */
  const Label = ({ label }) => {
    return (
      <label className="usa-label" htmlFor={htmlFor}>
        {label}
      </label>
    )
  }

  /**
   * a functional component to create a list of options for a select element.
   * @function
   * @param {array} options - The inherited options objects of value and label
   * @return {html} returns a semantic html option(s)
   */
  const Options = ({ options }) => {
    return options.map(option => {
      return (
        <option value={option.value} key={option.value}>
          {option.label}
        </option>
      )
    })
  }

  return (
    <>
      <Label label={label} />
      <select className="usa-select" name={htmlFor} id={htmlFor}>
        <Options options={options} />
      </select>
    </>
  )
}

export default Select
