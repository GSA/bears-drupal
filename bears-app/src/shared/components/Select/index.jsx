import { Label } from '../index'
import PropTypes from 'prop-types'
/**
 * a functional component to create a select from a list of options
 * @component
 * @param {string} label - The inherited options object of value and label
 * @param {string} htmlFor - The inherited component id that the label is for
 * @param {array} options - The inherited options array of objects with value and label keys
 * @return {html} returns a semantic html select element with options
 */
function Select({ label, htmlFor, options, defaultValue, onChange }) {
  /**
   * a functional component to create a list of options for a select element.
   * @function
   * @param {array} options - The inherited options objects of value and label
   * @return {html} returns a semantic html option(s)
   */
  const Options = ({ options }) => {
    return options.map(option => {
      return (
        <option
          value={option.value}
          key={option.value}
          selected={option.value === defaultValue}
        >
          {option.value}
        </option>
      )
    })
  }

  return (
    <>
      <Label label={label} htmlFor={htmlFor} />
      <select
        className="usa-select"
        name={htmlFor}
        id={htmlFor}
        onChange={onChange}
      >
        <Options options={options} />
      </select>
    </>
  )
}

Select.propTypes = {
  label: PropTypes.string,
  htmlFor: PropTypes.string,
  options: PropTypes.array,
}

export default Select
