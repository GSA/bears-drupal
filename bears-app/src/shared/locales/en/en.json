{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start",
      "list": [
        {
          "notice": "<div>Your information<strong>will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "heading": "Your potential benefits",
    "description": "<div>You must verify your eligibility and visit each agency to apply. <strong>Check with your city, state, or employer for other benefits.</strong></div>",
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    }
  },
  "notEligibleResults": {
    "heading": "Benefits you did not qualify for",
    "description": "<div>According to your answers you <strong>are not</strong> eligible for these benefits.</div>",
    "heading2": "Non-eligible Results",
    "description2": "Check with your state, city, or employer for other benefits."
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  }
}
