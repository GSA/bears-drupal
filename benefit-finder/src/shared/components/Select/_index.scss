@use '../../styles/colors/_index.scss' as color;

#benefit-finder {
  .usa-select {
    border-radius: 3px;

    &.usa-input--error {
      outline: 4px solid color.$alert-red !important;
      border: none;
    }
  }
}
