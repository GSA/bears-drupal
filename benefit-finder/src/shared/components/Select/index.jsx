import PropTypes from 'prop-types'
import { useHandleClassName } from '../../hooks/useHandleClassName'
import { Label } from '../index'

/**
 * a functional component to create a select from a list of options
 * @component
 * @param {string} label - The inherited options object of value and label
 * @param {string} htmlFor - The inherited component id that the label is for
 * @param {array} options - The inherited options array of objects with value and label keys
 * @param {string} selected - The inherited selected value
 * @param {function} onChange - The inherited onChange handler
 * @param {object} ui - The inherited object for ui translations
 * @param {string} className - inherited class string
 * @return {html} returns a semantic html select element with options
 */
function Select({
  label,
  htmlFor,
  options,
  selected,
  onChange,
  required,
  ui,
  className,
}) {
  const { labelSelect, defaultValue } = ui
  const handleRequired = required === 'TRUE' ? ['required-field'] : ''
  const defaultClasses = ['usa-select']
  const utilityClasses = handleRequired
  /**
   * a functional component to create a list of options for a select element.
   * @function
   * @param {array} options - The inherited options objects of value and label
   * @return {html} returns a semantic html option(s)
   */
  const Options = ({ options }) => {
    return options.map(option => {
      return (
        <option value={option.value} key={option.value}>
          {option.value}
        </option>
      )
    })
  }

  return (
    <>
      <Label label={label || labelSelect} htmlFor={htmlFor} />
      <select
        className={useHandleClassName({
          className,
          defaultClasses,
          utilityClasses,
        })}
        name={htmlFor}
        id={htmlFor}
        onChange={onChange}
        value={selected || ''}
      >
        <option value="" key="default" disabled>
          {defaultValue}
        </option>
        <Options options={options} />
      </select>
    </>
  )
}

Select.propTypes = {
  label: PropTypes.string,
  htmlFor: PropTypes.string,
  options: PropTypes.array,
  selected: PropTypes.string,
  onChange: PropTypes.func,
  ui: PropTypes.object,
  className: PropTypes.string,
}

export default Select
