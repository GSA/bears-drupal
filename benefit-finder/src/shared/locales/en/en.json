{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start:",
      "list": [
        {
          "notice": "<div>Your information<strong> will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "alertBanner": {
    "heading": "Error status",
    "description": "All fields marked required must be completed."
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "qualified": {
      "chevron": {
        "heading": "Your Custom List",
        "description": "<div><p><strong>Here is your list of potential Federal Benefits.</strong> Your state, city or employer may offer other benefits. Visit each agency providing the benefit to learn if you are truly eligible and conduct an actual application.</p></div>"
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you meet the criteria to qualify for these benefits:</strong></div>"
    },
    "notQualified": {
      "chevron": {
        "heading": "Benefits You Did Not Qualify For",
        "description": ""
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you did not meet the criteria to qualify for these benefits:</strong></div>"
    },
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    },
    "relativeBenefits": {
      "heading": "Other benefits that might be relevant to you"
    },
    "shareResults": {
      "heading": "Sharing and printing",
      "shareButton": "Share link",
      "emailButton": "Email results",
      "emailSubject": "Results from USAGovâ€™s benefit finder",
      "printButton": "Print results",
      "description": "Copy a link to this page with the criteria you selected above. Make sure to share it only with those you trust as your answers will be visible."
    }
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  },
  "sectionHeadings": {
    "start": "Complete Questions",
    "continue": "Questions Continued",
    "final": "Final Questions"
  },
  "select": {
    "labelSelect": "Select",
    "defaultValue": "Click to Select",
    "dateDefaultValue": "-Select-"
  },
  "date": {
    "alert": "Date should follow the format 01 - January 22 1977",
    "labelDay": "Day",
    "labelMonth": "Month",
    "labelYear": "Year",
    "monthOptions": [
      {
        "label": "01 - January"
      },
      {
        "label": "02 - February"
      },
      {
        "label": "03 - March"
      },
      {
        "label": "04 - April"
      },
      {
        "label": "05 - May"
      },
      {
        "label": "06 - June"
      },
      {
        "label": "07 - July"
      },
      {
        "label": "08 - August"
      },
      {
        "label": "09 - September"
      },
      {
        "label": "10 - October"
      },
      {
        "label": "11 - November"
      },
      {
        "label": "12 - December"
      }
    ]
  }
}
