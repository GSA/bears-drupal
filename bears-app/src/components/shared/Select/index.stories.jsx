import Select from './index.jsx'

const selectLabel = 'Dropdown Label'
const selectOptions = [
  { value: '', label: '-Select-' },
  { value: 'value-1', label: 'Option 1' },
  { value: 'value-2', label: 'Option 2' },
  { value: 'value-3', label: 'Option 3' },
]

export default {
  component: Select,
}

export const Primary = {
  render: () => (
    <Select label={selectLabel} options={selectOptions} htmlFor="options" />
  ),
}
