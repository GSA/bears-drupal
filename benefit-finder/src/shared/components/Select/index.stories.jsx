import Select from './index.jsx'
import * as en from '../../locales/en/en.json'

const selectOptions = [
  { value: 'value-1', label: 'Option 1' },
  { value: 'value-2', label: 'Option 2' },
  { value: 'value-3', label: 'Option 3' },
]

export default {
  component: Select,
  tags: ['autodocs'],
  args: {
    ui: en.select,
    options: selectOptions,
  },
}

export const Primary = {}
