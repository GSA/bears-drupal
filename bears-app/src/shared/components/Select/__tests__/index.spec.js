import TestRenderer from 'react-test-renderer'
import Select from '../index.jsx'
import * as en from '../../../locales/en/en.json'

const selectLabel = 'Dropdown Label'
const selectOptions = [
  { value: '', label: '-Select-' },
  { value: 'value-1', label: 'Option 1' },
  { value: 'value-2', label: 'Option 2' },
  { value: 'value-3', label: 'Option 3' },
]

describe('Select', () => {
  it('renders a match to the previous snapshot', () => {
    const component = TestRenderer.create(
      <Select label={selectLabel} options={selectOptions} ui={en.select} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
