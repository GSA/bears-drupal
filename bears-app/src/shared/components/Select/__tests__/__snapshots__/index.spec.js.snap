// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders a match to the previous snapshot 1`] = `
Array [
  <label
    className=""
  >
    Dropdown Label
  </label>,
  <select
    className="usa-select"
  >
    <option
      value=""
    >
      -Select-
    </option>
    <option
      value="value-1"
    >
      Option 1
    </option>
    <option
      value="value-2"
    >
      Option 2
    </option>
    <option
      value="value-3"
    >
      Option 3
    </option>
  </select>,
]
`;
